edb/server/args.py

```diff
@@ -109,8 +110,12 @@ def assign_implementation(self, cls):
     def compute_default_pool_size(self):
         if self.value == self.Fixed:
             return compute_default_compiler_pool_size()
-        else:
+        elif self.value == self.OnDemand:
             return 1
+        elif self.value == self.Remote:
+            return 2
```

Comment:
(1) Sure thing!

(2) 

> When the mode is set to remote, the pool size means how many concurrent compile requests can the EdgeDB server issue at most, default is `2`.

Well this doesn't really have a concrete reason, 2 is just something on top of my head.

Reply:
I still don't understand why is this `2` or for whatever it worth what any other number would even mean in this context.

Reply:
Before this PR, only 1 compile request can be submitted at the same time per UNIX connection. We will have only 1 UNIX connection between each EdgeDB server and the compiler server, so after this PR, multiple compile requests can "multiplex" on the same UNIX connection. In this case, `--compiler-pool-size` is used to set the max concurrency of the "multiplex", so that the compiler server won't be overloaded by a single EdgeDB server.